Require plan_check for WhatsApp block

stacimc opened the pull request:
<!--- Provide a general summary of your changes in the Title above -->
<!-- Would you like this feature to be tested by Beta testers?
Please add testing instructions to to-test.md in a new commit as part of your PR. -->

Fixes 6262-gh-Automattic/jpop-issues

#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
The `whatsapp-button` block is only meant to be inserted as a child of parent block `send-a-message`, and both should only be available on paid plans. This is managed by setting `plan_check` at block registration on `send-a-message`. You're able to circumvent the plan checks, however, by opening the code editor and removing the parent `send-a-message` block.

This PR closes that loophole by moving the `plan_check` from the parent down to the child.

\*\***This should not be merged into the `update/enable-frontend-upgrade-nudges` branch. I'll rebase on master before merging. The intent is to allow easier testing the block with the new upgrade nudges for now.**\*\*

#### Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

#### Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

#### Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

1. First apply D55983-code to your sandbox
2. Sync this branch to your sandbox with a8c-wp-env
3. Manually update the `wp-content/mu-plugins/jetpack/class.jetpack-gutenberg.php` file in your sandbox to match the [class.jetpack-gutenberg.php](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/class.jetpack-gutenberg.php) file in your local jetpack. _(You need this to test the upgrade nudges. This code has been merged to master but not deployed to wpcom yet, and the a8c-wp-env will not automatically sync this for you.)_

For Wpcom:
* On a free site, create a post and add a WhatsApp block.
* Verify that you see an upgrade nudge in the editor
* Open the post on the frontend and verify that you see the block on the frontend, with an upgrade nudge
* Return to the post and enter the Code Editor
* Remove the `send-a-message` block wrapping the WhatsApp block
* Exit the Code Editor and verify that you still see an upgrade nudge in the editor
* View the post in the frontend again and verify that you still see the upgrade nudge


#### Screenshots

**View of the block in the code editor -- highlighted code should be removed:**
<img width="1010" alt="Screen Shot 2021-01-21 at 4 59 56 PM" src="https://user-images.githubusercontent.com/63313398/105431389-604cb400-5c0a-11eb-8626-f91642675c7e.png">
<img width="1005" alt="Screen Shot 2021-01-21 at 5 00 08 PM" src="https://user-images.githubusercontent.com/63313398/105431395-62167780-5c0a-11eb-806a-f6eb40f87be3.png">

**Normal WhatsApp block in the editor:**
<img width="828" alt="Screen Shot 2021-01-21 at 4 46 15 PM" src="https://user-images.githubusercontent.com/63313398/105431583-bae61000-5c0a-11eb-97e3-b3852abc9f0d.png">

**WhatsApp block in the editor when the parent SendAMessage has been removed:**
<img width="848" alt="Screen Shot 2021-01-21 at 4 46 36 PM" src="https://user-images.githubusercontent.com/63313398/105431617-ca655900-5c0a-11eb-962f-df21e4dd2475.png">


**Blocks on the frontend:**
<img width="771" alt="Screen Shot 2021-01-26 at 1 17 15 PM" src="https://user-images.githubusercontent.com/63313398/105907199-8e027600-5fd9-11eb-8138-3187e88d0772.png">


#### Proposed changelog entry for your changes:
<!-- Please do not leave this empty. If no changelog entry needed, state as such. -->
<!-- Guidelines: https://github.com/Automattic/jetpack/blob/master/docs/writing-a-good-changelog-entry.md -->
*

stacimc committed:
[not verified] Move plan_check from Send-A-Message to WhatsApp

apeatling reviewed (dismissed):
This looks good. It would be great to add some space between the banner and the button on the front end if that's possible. Once you've got a dotcom patch I think this can go for external review.

stacimc commented:
I have opened D55983-code for updating the StoreProductList, but awaiting feedback on wpcom changes first. Moving this back to In Progress until that is verified.

stacimc committed:
[not verified] Prevent upgrade nudge from being clickable while hidden

The upgrade nudge banner is set to `opacity: 0` when hidden, which
makes the banner invisible but does not prevent it from being
clicked. This can be confusing because it can prevent clicks on the
underlying block, with no indication of what's happening.

`visibility: hidden` is chosen so that the banner still takes up
space when hidden, so that blocks do not jump around the page
excessively as the banners are hidden/unhidden.

stacimc commented:
Updated the spacing on the frontend and confirmed that D55983-code should be all we need. This should be ready for review again.

sdixon194 commented:
This tests well for me on my wpcom sandbox! 

I did run into an issue when testing on my Jetpack site. It looks like some of the CSS for the button is stripped. This issue wasn't present on the latest stable version and was applied when switching to this branch. Tested with multiple themes:

![Screen Shot on 2021-02-04 at 10:55:52](https://user-images.githubusercontent.com/33553323/106937121-630bd680-66eb-11eb-9341-97d9dbc3d505.png)

Again, just regular Jetpack sites - wpcom sites worked without an issue! 

This PR is a few weeks old - maybe it needs to be rebased? 

Thanks!

stacimc commented:
Thanks @sdixon194 - Rebased and the CSS is working for me on Jetpack now:

<img width="301" alt="Screen Shot 2021-02-04 at 2 29 06 PM" src="https://user-images.githubusercontent.com/63313398/106963436-6dcb6900-66f5-11eb-9a5c-5079b9c0945f.png">

The screenshot you posted is what it looks like when the block doesn't go through the `render_callback` (which loads the styles). I'm not sure what happened there 🤔

jeherve reviewed (dismissed):
This tests well for me. 🚢

stacimc commented:
Rebased as some of the tests were stuck.

jeherve commented:
@stacimc When do you think you'll be able to commit the matching WordPress.com diff? Thank you!

jeherve commented:
r221699-wpcom
